debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -101,6 +106,60 @@ public void resetSessionToCdb() {
         }
     }
 
+    public OracleDatabaseVersion getOracleVersion() {
+        if (databaseVersion == null) {
+            databaseVersion = resolveOracleDatabaseVersion();
+            LOGGER.info("Database Version: {}", databaseVersion.getBanner());
+        }
+        return databaseVersion;
+    }
+
+    private OracleDatabaseVersion resolveOracleDatabaseVersion() {
+        String versionStr;
+        try {
+            try {
+                // Oracle 18.1 introduced BANNER_FULL as the new column rather than BANNER
+                // This column uses a different format than the legacy BANNER.
+                versionStr = queryAndMap("SELECT BANNER_FULL FROM V$VERSION WHERE BANNER_FULL LIKE 'Oracle Database%'", (rs) -> {
+                    if (rs.next()) {
+                        return rs.getString(1);
+                    }
+                    return null;
+                });
+            }
+            catch (SQLException e) {
+                // exception ignored
+                if (e.getMessage().contains("ORA-00904: \"BANNER_FULL\": invalid identifier")) {
+                    LOGGER.debug("BANNER_FULL column not in V$VERSION, using BANNER column as fallback");
+                    versionStr = null;
+                }
+                else {
+                    throw e;
+                }
+            }
+
+            // For databases prior to 18.1, a SQLException will be thrown due to BANNER_FULL not being a column and
+            // this will cause versionStr to remain null, use fallback column BANNER for versions prior to 18.1.
+            if (versionStr == null) {
+                versionStr = queryAndMap("SELECT BANNER FROM V$VERSION WHERE BANNER LIKE 'Oracle Database%'", (rs) -> {
+                    if (rs.next()) {
+                        return rs.getString(1);
+                    }
+                    return null;
+                });
+            }
+        }
+        catch (SQLException e) {
+            throw new RuntimeException("Failed to resolve Oracle database version", e);
+        }
+
+        if (versionStr == null) {
+            throw new RuntimeException("Failed to resolve Oracle database version");
```

Comment:
So there's no way we can keep running without the version?

Reply:
LogMinerHelper methods that deal with redo and archive logs depend on very specific magic SCN values that are derived based on the current database version.  If we return a default version here, this would cause those methods that depend on this specific magic number to lead to the notorious `missing logfile` exception we've been trying to avoid.

I can look into moving away from these magic SCN value checks entirely, but we'd need to rework the code around fetching and working with both archive and redo logs, i.e. knowing which ones to add to the session, etc.

Reply:
Ok, I see. I can't really tell whether that's a feasible thing to look into or not. If you think it is, can you log it in Jira as a follow-up please, and we go with what we have for now.